pkg/login/client_login.go

```diff
@@ -45,6 +51,8 @@ func Login(config *config.Config, input io.Reader) error {
 		message := SuccessMessage(response.UserName, response.UserEmail, response.OrganizationName, response.TeamName, response.TeamMode == "console")
 		ansi.StopSpinner(s, message, os.Stdout)
 
+		config.Profile.SaveProfile(config.LocalConfigFile != "")
```

Comment:
I think we should provide error handling & also select this profile for subsequent requests?

```golang
	if err = config.Profile.SaveProfile(false); err != nil {
		return err
	}
	if err = config.Profile.UseProfile(); err != nil {
		return err
	}
```

Reply:
Good call on the error handling.

I've got a couple of questions:

1. Should passing a `--api-key` and or `--profile` to any command persist them? If so, should we centralize the handling of this? I notice the `ci` command has a similar piece of logic.
2. Should `--profile` take the `--config` value that has been passed into account? Noticed that I'm checking to see if it's set in order to work out if the profile should be saved to global or local. Should `UseProfile` also take a `local` flag?

Reply:
Yeah, tbh I don't have a good answer to your questions here. I think it's up to debate 😅 

> 1. Should passing a --api-key and or --profile to any command persist them? If so, should we centralize the handling of this? I notice the ci command has a similar piece of logic.

I don't think we should persist them normally, except for the `login` command in which case we should. But I'm not sure about that inconsistency. I'm trying to imagine use cases where people can be a part of multiple projects and even organizations, what is the most intuitive way for them to handle this?

I think the `--config` and `--local` flags are handled via the config handling, so the function `UseProfile` or `SaveProfile` will account for them implicitly.

Reply:
`UseProfile` always saves to the global config:

```go
func (p *Profile) UseProfile() error {
	p.Config.GlobalConfig.Set("profile", p.Name)
	return p.Config.WriteGlobalConfig()
}
```

Reply:
I think the thought process is that the idea of `profile` is less relevant for local config given that local config don't support profile and will override the profile values (API key, team ID) regardless.

Reply:
The only place that `SaveProfile(true)` is called is when `project use` is called and the `--local` flag is set. So, it is possible to set a profile in a local config.

It's a bit too much to try to get my head around all the permutations without opening up a can of worms 😊 So I think it's safer to stick to what you suggested, which maintains consistent behavior.

Reply:
Updates in https://github.com/hookdeck/hookdeck-cli/pull/101/commits/fb612d9ebfd13b32dd6680bdf1c149c66f56a62f